### DISCUSSION TITLE
Flatten an array in substation v1.1.0

### DISCUSSION THREAD

**Discussion by viraj-lunani** [brexhq/substation#155] [Q&A] [answered]
Hi @jshlbrd  How can I flatten an array in substation v1.1.0. Similar to [this](https://substation.readme.io/v0.9.1/docs/flatten) processor in v0.9.1
My situation: I have a an array in a field called 'a', and I have another array in a field called 'b'. I want to add all elements from 'a' to 'b'. 
e.g.
'a': ['val1', 'val2']
'b': ['val3', 'val4']
using for_each I can get 
'b': ['val3', 'val4', ['val1', 'val2']]
How can I flatten this array. Or better yet, is there a transform to "extend array" as I am trying to do.

**Comment by jshlbrd** [comment, marked as answer]
Hey @viraj-lunani, the flatten processor wasn't migrated to v1.0 since it's supported natively by Substation's JSON parser (GJSON). You can flatten `{"b":["val3","val4",["val1","val2"]]}` by using a copy transform and the GJSON `@flatten` modifier. Try this:
```jsonnet
sub.tf.object.copy({object: {source_key: 'b|@flatten', target_key: 'b'}})
```

If you need to extend often, then you could build a custom transform Jsonnet function like:
```jsonnet

extend(source, target): [
  // This appends the value from `source` to the end of `target`.
  sub.tf.object.copy({object: {source_key: source, target_key: target + '.-1' }}),
  sub.tf.object.copy({object: {source_key: target + '|@flatten', target_key: target}}),  
],
```

If you want to try GJSON in the browser, then you can use [their playground site](https://gjson.dev/).

The flatten modifier works for valid JSON without keys as well (e.g. `["val3","val4",["val1","val2"]]`) if you chain the `@this` modifier into it, like:

```jsonnet
sub.tf.object.copy({object: {source_key: '@this|@flatten'}})
```

**Reply by viraj-lunani** [reply]
Awesome, thanks!

**Reply by jshlbrd** [reply]
No problem. If you're interested in making a PR, I think these would be good config examples to add to the repo.

This folder would be a good place to add them: https://github.com/brexhq/substation/tree/main/examples/config/transform/array.
